src/main/java/pl/betoncraft/betonquest/compatibility/protocollib/NPCHider.java

```diff
@@ -132,11 +132,8 @@ public void applyVisibility(final Player player, final Integer npcID) {
         if (conditions == null || conditions.isEmpty()) {
             hidden = false;
         } else {
-            for (final ConditionID condition : conditions) {
-                if (!BetonQuest.condition(PlayerConverter.getID(player), condition)) {
-                    hidden = false;
-                    break;
-                }
+            if (!BetonQuest.conditions(PlayerConverter.getID(player), conditions)) {
+                hidden = false;
```

Comment:
Can be merged with previous if-branch and then directly used to set `hidden`. As hidden also is only used once it might then be moved to into the if condition that until now reads `if (hidden)`:

```java
            if (conditions == null || conditions.isEmpty() || !BetonQuest.conditions(PlayerConverter.getID(player), conditions)) {
                hider.showEntity(player, npc.getEntity());
            } else {
                hider.hideEntity(player, npc.getEntity());
            }
```